### PR TITLE
Separate determining FQDN, add tests, make FQDN regex optional

### PR DESCRIPTION
**Pull request by KevinMGranger**
This relates to #47.

One very notably change is the introduction of Autoloading and Namespacing. Those are tenets of modern, best-practices PHP, but does entail running `composer install` before the project is usable.

I don't think that's too much to ask, considering there's already a composer.phar included in the repo (although I'd prefer everyone had their own copy of composer).

Thoughts?

### CODE REVIEW

File: phplib/base.php [hunk 1]
```diff
@@ -30,10 +33,8 @@
 $sleep_state_levels = array(-1 => "Unknown", 1 => "NREM Stage 1", 2 => "NREM Stage 2", 3 => "NREM Stage 3", 4 => "REM");
 
 
-// Test to make sure we can handle this team
-// .. and handle dev
-$fqdn = (isset($fqdn)) ? $fqdn : $_SERVER['HTTP_HOST'];
-$fqdn = preg_replace($dev_fqdn, $prod_fqdn, $fqdn);
+// Test to make sure we can handle this team.
+$fqdn = (new FQDN(isset($fqdn) ? $fqdn : $_SERVER['HTTP_HOST'], $dev_fqdn, $prod_fqdn))->getFQDN();
```
Comment (mrtazz) [inline phplib/base.php]: could you unpack this statement a bit? This is pretty hard to parse with the ternary statement in there.

File: src/FQDN.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
+<?php
+namespace Etsy\Opsweekly;
+
+use \InvalidArgumentException;
```
Comment (mrtazz) [inline src/FQDN.php]: not sure why this is included. Is this thrown somewhere implicitly? Should it be caught somewhere as well then?

**Comment by mrtazz** [comment]
Thank you for the contribution! I left a couple of comments on the code, but overall this looks great.

File: src/FQDN.php (hunk 2 above)
Comment (KevinMGranger) [inline src/FQDN.php]: I was originally trying to handle errors with the regex, but it turned out to be tricky to test for. I'll remove it.

**Comment by KevinMGranger** [comment]
Changes made!

Thoughts on ./composer.phar in the README, versus telling people to install composer on their own? (And removing composer.phar) ?

**Comment by mrtazz** [comment]
the changes look good to me now. I think it makes sense to have everyone install their own composer and not track this in here. But I don't feel super strongly about this. We can discuss it in a separate pull request to not block this one with this question. @lozzd did you have any additional comments on this change?
